src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -380,6 +446,113 @@ class _Call21Bool(Protocol):
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (scalar, scalar) -> bool
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.bool: ...
+    @overload  # (array-like, array, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (array-like, array, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array, array-like, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (array, array-like, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array-like, array) -> Array[?]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        out: _Out1[_AnyArray] | None = None,
+        *,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array, array-like) -> Array[?]
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        out: _Out1[_AnyArray] | None = None,
+        *,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array-like, array-like, out: T) -> T
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: ArrayLike,
+        /,
+        out: _ArrayT | tuple[_ArrayT],
+        *,
+        dtype: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> _ArrayT: ...
+    @overload  # (array-like, array-like) -> Array[?] | ?
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _ArrayLikeNumber_co,
+        /,
+        out: _Out1[_AnyArray] | None = None,
+        *,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool] | np.bool: ...
```

Comment:
This might also return `NDArray[np.object_] | bool`. But including that would be messy, so it's better to just `Any`. This is the last overload, so it won't be matched very often (or at least, that's the usually the idea behind a *fallback* overload like this one).